src/clipboard.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016 Joe Wilm, The Alacritty Project Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(any(target_os = "linux", target_os = "bsd"))]
+use clipboard::x11_clipboard::{X11ClipboardContext, Clipboard as X11SecondaryClipboard};
+use clipboard::{ClipboardProvider, ClipboardContext};
+
+#[derive(Debug)]
+pub enum Clipboard {
+    Primary,
+    Secondary,
+}
+
+impl Clipboard {
+    pub fn store(&self, text: impl Into<String>) {
```

Comment:
I think I'd prefer to return a `Result` here and log warning messages and such where this is used.

Reply:
We never need to handle the error case but only want to log it. This effectively just reduces code bloat by having the same code at all callsites.

Since this is for internal use only, not a public API, I think this is the better approach here.

Reply:
Makes sense then